src/nbl/video/utilities/IUtilities.cpp

```diff
@@ -837,6 +837,7 @@ void IUtilities::updateImageViaStagingBuffer(
 
     ImageRegionIterator regionIterator = ImageRegionIterator(regions, queueFamProps, srcBuffer, /*TODO*/dstImage->getCreationParameters().format, dstImage);
 
+    // TODO[FUTURE]: consider benefits of using `limits->optimalBufferCopyOffsetAlignment` and `limits->optimalBufferCopyRowPitchAlignment`
```

Comment:
`optimalBufferCopyOffsetAlignment` you'd use when allocating memory from the staging buffer for the output image region

`optimalBufferCopyRowPitchAlignment` you'd use to step the output image's stride to something custom/rounded up

Reply:
in all utilities do
```cpp
const uint32_t allocationAlignment = core::max<uint32_t>(limits.nonCoherentAtomSize,limits.optimalBufferCopyOffsetAlignment);
```

but then all the `assert` that check stuff on `nonCoherentAtomSize` should be checking the above value instead.

Reply:
`optimalBufferCopyRowPitchAlignment` is something to handle in your `ImageRegionIterator`

Reply:
> in all utilities do
> 
> ```c++
> const uint32_t allocationAlignment = core::max<uint32_t>(limits.nonCoherentAtomSize,limits.optimalBufferCopyOffsetAlignment);
> ```
> 
> but then all the `assert` that check stuff on `nonCoherentAtomSize` should be checking the above value instead.

the name is a bit misleading @devshgraphicsprogramming, this is only related to image<->buffer  copies

Reply:
well then pick a different name for `allocationAlignment`